pkg/flow/adapter/xslttransform/adapter.go

```diff
@@ -142,14 +142,8 @@ func (a *xsltTransformAdapter) dispatch(ctx context.Context, event cloudevents.E
 			fmt.Errorf("eror processing XML with XSLT: %v", err), nil)
 	}
 
-	if isXML {
-		if err := event.SetData(cloudevents.ApplicationXML, []byte(output)); err != nil {
-			return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
-		}
-	} else {
-		if err := event.SetData(cloudevents.ApplicationJSON, []byte(output)); err != nil {
-			return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
-		}
+	if err := event.SetData(cloudevents.ApplicationXML, []byte(output)); err != nil {
```

Comment:
I really still don't get how this is relevant _outside_ of `if sink` below, because now this affects _both_ modes and I don't think we want that.

Does it mean the transformer didn't work at all before the sink feature? Hard for me to believe.

Again, asking for @odacremolbap's help to clarify because this part of the code smells to me.

Reply:
you are right!